### PR TITLE
[dotnet] Deprecate saving screenshots with different image formats

### PR DESCRIPTION
**Pull request by nvborisenko**
### Description
This is the 1st phase of out journey to remove `System.Drawing.Common` dependency.

### Motivation and Context
Any great library should keep number of dependencies minimized as possible.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by nvborisenko** [comment]
Enabler for https://github.com/SeleniumHQ/selenium/pull/12781

File: dotnet/src/webdriver/Screenshot.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ namespace OpenQA.Selenium
     /// <summary>
     /// File format for saving screenshots.
     /// </summary>
+    [Obsolete("Selenium doesn't support decoding/encoding between different image formats.")]
```
Comment (titusfortner) [inline dotnet/src/webdriver/Screenshot.cs]:
```suggestion
    [Obsolete("Support for decoding/encoding between different image formats is deprecated")]
```

File: dotnet/src/webdriver/Screenshot.cs [hunk 2]
```diff
@@ -84,6 +85,7 @@ public override void SaveAsFile(string fileName)
         /// <param name="fileName">The full path and file name to save the screenshot to.</param>
         /// <param name="format">A <see cref="ScreenshotImageFormat"/> value indicating the format
         /// to save the image to.</param>
+        [Obsolete("Selenium doesn't support decoding/encoding between different image formats. Use \"SaveAsFile(string fileName)\" method instead.")]
```
Comment (titusfortner) [inline dotnet/src/webdriver/Screenshot.cs]:
```suggestion
        [Obsolete("Support for decoding/encoding between different image formats is deprecated; use SaveAsFile(string fileName) method instead")]
```

**Review by titusfortner — APPROVED** [review]
I offered suggestions that match how we say it in Ruby, feel free to keep or change.

**Pull request merged**